xblock/plugin.py

```diff
@@ -41,6 +41,32 @@ def default_select(identifier, all_entry_points):  # pylint: disable=inconsisten
         raise AmbiguousPluginError(all_entry_points)
 
 
+def select_with_overrides(identifier, all_entry_points):
+    """
+    Honors the ability for an XBlock to override the default entry point.
+    Raise an exception when we have ambiguous entry points.
+    """
+
+    # Split entry points into overrides and non-overrides
+    overrides = []
+    block_entry_points = []
+
+    for block_entry_point in all_entry_points:
+        if block_entry_point.group.endswith('overrides'):
+            overrides.append(block_entry_point)
+        else:
+            block_entry_points.append(block_entry_point)
+
+    # Overrides get priority
+    if len(overrides) == 1:
+        return overrides[0]
+    elif len(overrides) > 1:
+        raise AmbiguousPluginError(all_entry_points)
+
+    # ... then default behavior
+    return default_select(identifier, block_entry_points)
```

Comment:
IMO a good defensive behavior would be to raise PluginMissingError in the case where an `xblock.v1.overrides` entry exists but the corresponding regular `xblock.v1` doesn't exist. You could achieve that here by performing default selection before checking the overrides list.

My thinking on this is that it would be weird to try to override something that doesn't exist in the first place, and the operator would probably want to be alerted to that weirdness. If I'm wrong, then we could always relax this assumption in the future without breaking anyone, whereas tightening the assumption in the future would be a breaking change.